cms/static/js/views/active_video_upload_list.js

```diff
@@ -59,6 +69,162 @@ define([
+            },
+
+            getProviderPlan: function() {
+                return this.availableTranscriptionPlans;
+            },
+
+            getTurnaroundPlan: function() {
+                if (this.selectedProvider){
+                    return this.availableTranscriptionPlans[this.selectedProvider].turnaround;
+                }
+            },
+
+            getFidelityPlan: function() {
+                if (this.selectedProvider == 'Cielo24') {
+                    return this.availableTranscriptionPlans[this.selectedProvider].fidelity;
+                }
+            },
+
+            getPlanLanguages: function() {
+                if (this.selectedProvider){
+                    var selectedPlan = this.availableTranscriptionPlans[this.selectedProvider];
+                    if (this.selectedProvider == 'Cielo24') {
+                        return selectedPlan.fidelity[this.selectedFidelityPlan].languages;
+                    }
+                    return selectedPlan.languages;
+                }
+            },
+
+            fidelitySelected: function(event) {
+                this.selectedFidelityPlan = event.target.value;
+                this.manageLanguageContainer();
+            },
+
+            turnaroundSelected: function(event) {
+                this.selectedTurnaroundPlan = event.target.value;
+                this.manageLanguageContainer();
+            },
+
+            providerSelected: function(event) {
+                this.selectedProvider = event.target.value;
+                this.populatePreferenceOptions();
+            },
+
+            manageLanguageContainer: function() {
+                var isTurnaroundSelected = this.$el.find('#transcript-turnaround')[0].options.selectedIndex,
+                    isFidelitySelected = this.$el.find('#transcript-fidelity')[0].options.selectedIndex;
+
+                if ((isTurnaroundSelected > 0 && this.selectedProvider === '3PlayMedia') || (isTurnaroundSelected  > 0 && isFidelitySelected > 0)) {
+                    this.availableLanguages = this.getPlanLanguages();
+                    this.$el.find('.transcript-languages-action-wrapper').show();
+                } else {
+                    this.availableLanguages = {};
+                    this.$el.find('.transcript-languages-action-wrapper').hide();
+                }
+            },
+
+            setTranscriptData: function(){
+                if (this.activeTranscriptionPlan) {
+                    this.selectedProvider = this.activeTranscriptionPlan['provider'];
+                    this.selectedFidelityPlan = this.activeTranscriptionPlan['cielo24_fidelity'];
+                    this.selectedTurnaroundPlan = this.activeTranscriptionPlan['cielo24_turnaround'] ? this.activeTranscriptionPlan['cielo24_turnaround']: this.activeTranscriptionPlan['three_play_turnaround'];
+                    this.activeLanguages = this.activeTranscriptionPlan['preferred_languages'];
```

Comment:
**TODO**: May need to deserialise this as it is probably received like this format `"[u'ru', u'en']"`